entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/response/getter/ArrayGetter.java

```diff
@@ -43,53 +43,74 @@ public Value getValue(final JsonNode jsonNode) throws ConversionException {
 
     while (elements.hasNext()) {
       final JsonNode node = elements.next();
+      boolean valueSet = false;
+
+      for (ValueGetter getter : rootGetters) {
+        if (getter.matches(node)) {
+          final Value value;
+          value = getter.getValue(node);
+          values.add(value);
+          valueSet = true;
+          break;
+        }
+      }
 
-      final Value value;
-
-      if (nestedValueGetter.matches(node)) {
-        value = nestedValueGetter.getValue(node);
-      } else if (directValueGetter.matches(node)) {
-        value = directValueGetter.getValue(node);
-      } else {
+      if (!valueSet) {
         throw new ConversionException(String.format("Unexpected node (%s) found", node));
       }
-
-      values.add(value);
     }
 
-    final ValueType primitiveType =
-        values.stream().map(Value::getValueType).findFirst().orElse(STRING);
-    final ValueType type = valueHelper.getArrayValueType(primitiveType);
-
-    final Value.Builder valueBuilder = Value.newBuilder().setValueType(type);
+    final ValueType type = values.stream().map(Value::getValueType).findFirst().orElse(STRING);
 
+    final Value.Builder valueBuilder = Value.newBuilder();
     switch (type) {
-      case STRING_ARRAY:
+      case STRING:
         values.stream().map(Value::getString).forEach(valueBuilder::addStringArray);
+        valueBuilder.setValueType(STRING_ARRAY);
         break;
 
-      case INT_ARRAY:
+      case INT:
         values.stream().map(Value::getInt).forEach(valueBuilder::addIntArray);
+        valueBuilder.setValueType(INT_ARRAY);
         break;
 
-      case LONG_ARRAY:
+      case LONG:
         values.stream().map(Value::getLong).forEach(valueBuilder::addLongArray);
+        valueBuilder.setValueType(LONG_ARRAY);
         break;
 
-      case FLOAT_ARRAY:
+      case FLOAT:
         values.stream().map(Value::getFloat).forEach(valueBuilder::addFloatArray);
+        valueBuilder.setValueType(FLOAT_ARRAY);
         break;
 
-      case DOUBLE_ARRAY:
+      case DOUBLE:
         values.stream().map(Value::getDouble).forEach(valueBuilder::addDoubleArray);
+        valueBuilder.setValueType(DOUBLE_ARRAY);
         break;
 
-      case BYTES_ARRAY:
+      case BYTES:
         values.stream().map(Value::getBytes).forEach(valueBuilder::addBytesArray);
+        valueBuilder.setValueType(BYTES_ARRAY);
         break;
 
-      case BOOLEAN_ARRAY:
+      case BOOL:
         values.stream().map(Value::getBoolean).forEach(valueBuilder::addBooleanArray);
+        valueBuilder.setValueType(BOOLEAN_ARRAY);
+        break;
+
+      case INT_ARRAY:
+      case LONG_ARRAY:
+      case STRING_ARRAY:
+      case FLOAT_ARRAY:
+      case DOUBLE_ARRAY:
+      case BYTES_ARRAY:
+      case BOOLEAN_ARRAY:
+      case STRING_MAP:
+      case VALUE_MAP:
```

Comment:
Dont have the complete context but this doesn't seem to be correct. Why is ValueArray being set in the case of VALUE_MAP?

Reply:
Basically, we are checking the "containing type".
If the containing type is anything other than a primitive, the current type is a nested array (represented as "VALUE_ARRAY").

@bhubam For clarity, could you rename the variable to "containingType" in line 63?